src/mpi/datatype/datatype_impl.c

```diff
@@ -36,15 +36,93 @@ void MPIR_Get_count_impl(const MPI_Status * status, MPI_Datatype datatype, MPI_A
             (*count) = 0;
         }
     }
+    return MPI_SUCCESS;
 }
 
-void MPIR_Pack_size_impl(int incount, MPI_Datatype datatype, MPI_Aint * size)
+int MPIR_Pack_impl(const void *inbuf, MPI_Aint incount, MPI_Datatype datatype,
+                   void *outbuf, MPI_Aint outsize, MPI_Aint * position, MPIR_Comm * comm_ptr)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPI_Aint actual_pack_bytes;
+    void *buf = (void *) ((char *) outbuf + *position);
+    mpi_errno = MPIR_Typerep_pack(inbuf, incount, datatype, 0, buf, outsize, &actual_pack_bytes);
+    if (mpi_errno)
+        goto fn_fail;
```

Comment:
nitpick: should this be `MPI_ERR_CHECK(mpi_errno)`?

Reply:
Oh, I see this was just copy/paste from before. No need to fix now.

Reply:
Thanks. I'll fix it before merging.